chain-tx-enclave-next/tx-query-next/enclave-app/src/sgx_module/handler/decryption_request.rs

```diff
@@ -31,36 +31,41 @@ pub fn verify_decryption_request(decryption_request: &DecryptionRequest, challen
 
 pub fn handle_decryption_request(
     decryption_request: &DecryptionRequest,
-    zmq_stream: Arc<Mutex<TcpStream>>,
+    chain_data_stream: Arc<Mutex<TcpStream>>,
 ) -> Result<DecryptionResponse, String> {
     // Prepare enclave request
     let enclave_request = EnclaveRequest::GetSealedTxData {
         txids: decryption_request.body.txs.clone(),
     }
     .encode();
 
-    let mut zmq_stream = zmq_stream.lock().unwrap();
+    let mut chain_data_stream = chain_data_stream.lock().unwrap();
 
-    // Send request to ZeroMQ
-    zmq_stream
+    // Send request to chain-abci
+    chain_data_stream
         .write_all(&enclave_request)
-        .map_err(|err| format!("Error while writing request to ZeroMQ: {}", err))?;
+        .map_err(|err| format!("Error while writing request to chain-abci: {}", err))?;
 
-    // Read reponse length from ZeroMQ (little endian u32 bytes)
+    // Read reponse length from chain-abci (little endian u32 bytes)
     let mut response_len = [0u8; 4];
-    zmq_stream
-        .read(&mut response_len)
-        .map_err(|err| format!("Error while reading reponse length from ZeroMQ: {}", err))?;
+    chain_data_stream.read(&mut response_len).map_err(|err| {
```

Comment:
how about
assert!(4==  chain_data_stream.read(&mut response_len)?..);

Reply:
it may be easier to use https://github.com/crypto-com/chain/pull/2108#discussion_r469074322